packages/router-generator/src/generator.ts

```diff
@@ -312,9 +332,40 @@ export async function generator(config: Config) {
       return
     }
 
+    node.isVirtualParentRequired =
+      removeGroups(node.path ?? '')
+        .split('')
+        .filter((char) => char === '/').length >= 2 && // check that the parent route wouldn't be the root route
+      !node.parent &&
+      node.isLayout
+
+    if (!node.isVirtual && node.isVirtualParentRequired) {
+      const parentNode = {
+        ...node,
+        path: removeLastSegmentFromPath(node.path) || '/',
+        filePath: removeLastSegmentFromPath(node.filePath) || '/',
+        fullPath: removeLastSegmentFromPath(node.fullPath) || '/',
+        routePath: removeLastSegmentFromPath(node.routePath) || '/',
+        variableName: removeLastSegmentFromVariableName(node.variableName),
+        isVirtual: true,
+        isLayout: false,
+        isVirtualParentRoute: true,
+        isVirtualParentRequired: false,
+      }
+
+      parentNode.children = parentNode.children ?? []
+      parentNode.children.push(node)
+
+      node.parent = parentNode
+
+      await handleNode(parentNode)
```

Comment:
can it be a problem that this parentNode is created "on-the-fly"?
e.g. what happens if two routes need the same virtual parent?